### PR TITLE
Docs extproc 1.17 changes

### PR DESCRIPTION
**Pull request by Nadine2016**
# Description

Please include a high level summary of the changes.

This bug fixes ... \ This new feature can be used to ...

_Fill out any of the following sections that are relevant and remove the others_
## API changes
- Added x field to y resource
- ...

## Code changes
- Fix error in `Foo()` function
- Add `Bar()` function
- ...

## CI changes
- Adjusted schedule for x job
- ...

## Docs changes
- Added guide about feature x to public docs
- Updated README to account for y behavior
- ...

# Context

Users ran into this bug doing ... \ Users needed this feature to ...

See slack conversation [here](https://solo-io-corp.slack.com/archives/some/post)

## Interesting decisions
 
We chose to do things this way because ...

## Testing steps

I manually verified behavior by ...

## Notes for reviewers

Be sure to verify intended behavior by ...

Please proofread comments on ...

This is a complex PR and may require a huddle to discuss ...

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: docs/content/guides/traffic_management/extproc/about.md [hunk 1]
```diff
@@ -36,6 +36,14 @@ The ExtProc server is a gRPC interface that must be able to respond to events in
...
 
+{{% notice warning %}}
+In Gloo Edge version 1.17.0, the Gloo Edge extProc filter implementation was changed to comply with the latest extProc implementation in Envoy. Previously, request and response attributes were included only in a [header processing request](https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/ext_proc/v3/external_processor.proto#service-ext-proc-v3-httpheaders) and therefore sent to the extProc server only when request header processing messages were configured to be sent. Starting in Gloo Edge version 1.17.0, the Gloo extProc filter sends request and response attributes as part of the top level [processing request](https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/ext_proc/v3/external_processor.proto#service-ext-proc-v3-processingrequest). That way, attributes can be processed on the first processing request regardless of its type.  </br></br>
+
+If you implemented your extProc server to expect request and response attributes as part of the HTTP header processing request, you must change this implementation to read attributes from the top level processing request instead. </br></br>
```
Comment (Rachael-Graham) [inline docs/content/guides/traffic_management/extproc/about.md:42]:
```suggestion
If you implemented your extProc server to expect request and response attributes as part of the HTTP header processing request, you must change this implementation to read attributes from the top-level processing request instead. </br></br>
```

File: docs/content/operations/upgrading/faq.md [hunk 2]
```diff
@@ -41,10 +41,33 @@ You might deploy Gloo Edge in Kubernetes environments that use the Kubernetes lo
...
 
 Review the following changes made to Gloo Edge in version {{< readfile file="static/content/version_geoss_latest_minor.md" markdown="true">}}. For some changes, you might be required to complete additional steps during the upgrade process.
 
+--> 
+
+### Breaking changes 
+
+#### ExtProc attribute processing {#extproc}
+
+The Gloo Edge extProc filter implementation was changed to comply with the latest extProc implementation in Envoy. Previously, request and response attributes were included only in a [header processing request](https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/ext_proc/v3/external_processor.proto#service-ext-proc-v3-httpheaders) and therefore sent to the extProc server only when request header processing messages were configured to be sent. Starting in Gloo Edge version 1.17.0, the Gloo extProc filter sends request and response attributes as part of the top level [processing request](https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/ext_proc/v3/external_processor.proto#service-ext-proc-v3-processingrequest). That way, attributes can be processed on the first processing request regardless of its type.
```
Comment (Rachael-Graham) [inline docs/content/operations/upgrading/faq.md:56]: same comments

File: docs/content/operations/upgrading/faq.md [hunk 3]
```diff
@@ -41,10 +41,33 @@ You might deploy Gloo Edge in Kubernetes environments that use the Kubernetes lo
...
+If you implemented your extProc server to expect request and response attributes as part of the HTTP header processing request, you must change this implementation to read attributes from the top level processing request instead. 
+
+For more information, see the [extProc proto definition](https://github.com/envoyproxy/envoy/blob/main/api/envoy/service/ext_proc/v3/external_processor.proto) in Envoy. 
+
+#### Envoy version 1.29 upgrade
+
+The Envoy dependency in Gloo Edge 1.17 was upgraded from 1.27.x to 1.29.x. This upgrade includes the following changes. For more information about these changes, see the [Envoy changelog documentation](https://www.envoyproxy.io/docs/envoy/latest/version_history/v1.29/v1.29).
+
+* **ExtProc attribute processing**: For more information, see [ExtProc attribute processing](#extproc).
+* **JWT tokens**: The behavior for extracting JWT tokens changed. Previously, the JWT token was cut in to non-base64 characters. Now, the entire JWT token is passed for validation. This change can be reverted temporarily by setting `envoy.reloadable_features.token_passed_entirely` to `false`.
```
Comment (Rachael-Graham) [inline docs/content/operations/upgrading/faq.md:67]:
```suggestion
* **JWT tokens**: The behavior for extracting JWT tokens changed. Previously, the JWT token was cut into non-base64 characters. Now, the entire JWT token is passed for validation. This change can be reverted temporarily by setting `envoy.reloadable_features.token_passed_entirely` to `false`.
```

File: docs/content/operations/upgrading/faq.md [hunk 4]
```diff
@@ -41,10 +41,33 @@ You might deploy Gloo Edge in Kubernetes environments that use the Kubernetes lo
...
+* **HTTP2 host header**: The HTTP2 host header is discarded if the `:authority` header is received. This change makes Envoy compliant with the HTTP2 request pseudo-header field implementation. For more information, see the [HTTP2 reference](https://www.rfc-editor.org/rfc/rfc9113#section-8.3.1). You can temporarily revert this change by setting the `envoy.reloadable_features.http2_discard_host_header` runtime flag to `false`.
+* **Transfer encoding header**: THe transfer encoding header is removed from downstream request headers. You can temporarily revert thsi change by setting `envoy.reloadable_features.sanitize_te` to `false`.
```
Comment (Rachael-Graham) [inline docs/content/operations/upgrading/faq.md:69]:
```suggestion
* **Transfer encoding header**: The transfer encoding header is removed from downstream request headers. You can temporarily revert this change by setting `envoy.reloadable_features.sanitize_te` to `false`.
```

**Review by Rachael-Graham — APPROVED** [review]

**Review by artberger — APPROVED** [review]

**Review by Rachael-Graham — APPROVED** [review]

**Pull request merged**